### PR TITLE
Reduce emails going to spam box by improving spamassassin score by including plain text version along with html version.

### PR DESCRIPTION
**Pull request by Jilt6502**

### CODE REVIEW

**Comment by Jilt6502** [comment]
When setting up pH7, I found that sometimes the notification emails
went to my spam box. Partly it was due to me not having properly
configured DKIM and SPF records on my DNS records. Actually I'm
thankful I had those improperly set because otherwise I may not have
noticed that the spam score was also affected by lacking a plain text
section. See https://gist.github.com/tylermakin/d820f65eb3c9dd98d58721c7fb1939a8
for more details including a section for smart watches (which I did
not attempt to implement since I don't have one and it does not
currently adversely affect spam score).
My first thought (wrong) was to make many changes throughout many
files to create the text version. Then it dawned on me to just convert
the html body into plain text and use that to put into the multipart
mime email. In looking for some way to convert to text I found various
examples and some that were security risks or that just didn't work
well. I am not qualified to state that the html2text code I opted to
use in my patch is secure but it does appear to do a good job at
making spamassassin  happy. I used the service at http://isnotspam.com
to check the spam score as I made changes. The emails still receive a
bad score for "BAYES_99 BODY: Bayes spam probability is 99 to 100%"
and just a tiny bit bad score for "BAYES_999 BODY: Bayes spam
probability is 99.9 to 100%" before my patch a bad score was also
received for "MPART_ALT_DIFF BODY: HTML and text parts are different"
and the suggestion I found was to Include a plain-text part to your
HTML message. There is a possibility for further improvement to the
score by verifying that the html part of the message validates with
validator.w3.org. I have not yet tested the spam score of emails that
are being sent if a person lacks swiftmailer.

**Comment by Jilt6502** [comment]
"before my patch a..." should start a new sentence "Before my patch a..."

**Comment by polynamaude** [comment]
@Jilt6502 Not to break much of your joy. Maybe this ain't the best software if your expectation are ending up with something "top notch". If you read thru some of the closed issue, many of them never got fixed.

This look much more like a "one man" project than anything else. I gave it a try about two years ago and went on to something else. However, I continued to get the emails regarding new issues and I often take the time trying to help people with the knowledge I have.

I'll give you a example, not so long ago (less than 6 month), the software had a "feature" that made it back link (that is show publicity of another site, in this case, the one of the author) thru your own website. This impact of your Google score and is simply a bad practice. What it does is also trying to make the author website look like more popular in Google Ranks because many other site relate to it. This wasn't in any of the documentation but many people who tried the software themselves removed it but just no one had ever took the time to make a little bit of a public trial on the matter. Another example would be that if you look at your Web pages (thru the browser), you'll see copyright before the HTML code, this again make your website being refused on AdSense because of improper language coding. And I could give you plenty more on this...

Yeah, look good on the outside but it's more of a "catch" for people who want a quick solution helping build a dating website (in a over crowded market) and get rich instaniously without much work... But you'll soon discover that there's more than a handful of bugs....

**Comment by polynamaude** [comment]
https://github.com/pH7Software/pH7-Social-Dating-CMS/issues/781#issue-627967333

File: composer.json [hunk 1]
```diff
@@ -63,7 +63,8 @@
     "twilio/sdk": "^5.28",
     "arcturial/clickatell": "^3.0",
     "fzaninotto/faker": "^1.8",
-    "guzzlehttp/guzzle": "^6.3"
+    "guzzlehttp/guzzle": "^6.3",
+    "kranemora/html2text": "^1.0"
```
Comment (pH-7) [inline composer.json:67]: Can you also commit package.lock with the changes of this package?

File: _protected/framework/Mail/Mail.class.php [hunk 2]
```diff
@@ -49,6 +50,9 @@ public function send(array $aInfo, $sContents, $iFormatType = Mailable::HTML_FOR
             ->setTo([escape($sToMail, true) => escape($sToName, true)]);
 
         if ($iFormatType === Mailable::HTML_FORMAT) {
+            $html2Text = new \kranemora\Html2Text\Html2Text;
```
Comment (pH-7) [inline _protected/framework/Mail/Mail.class.php:53]: Have you tested it? How it went?

**Comment by polynamaude** [comment]
@pH-7 Thanks for nicely integrating others contribution. You are the best !

**Pull request closed without merging**